### PR TITLE
Add installation instructions for less experienced Python users.

### PR DESCRIPTION
**Pull request by j-bennet**
As a Python developer, I found eland easy to install, but I think the instructions can be improved for users that didn't deal with installing Python packages before.

### CODE REVIEW

**Review by davidkyle — COMMENTED** [review]
Thanks for the PR @j-bennet. I left a few comments but I think is this good and you've high lighted some key points.

File: README.md [hunk 1]
```diff
@@ -69,14 +106,26 @@ $ sudo apt-get install -y \
 ```
 
 Note that other distributions such as CentOS, RedHat, Arch, etc. may require using a different package manager and
-specifying different package names. 
+specifying different package names.
+
+If you're using Windows, it's likely that you don't need to install any of those.
+
+If you're using macOS, you might need to install Xcode command-line tools:
```
[the remark this answers is not in the document]
Comment (j-bennet) [reply README.md:113]: xcode command-line tools will be required to install Python, if it's old or missing. I should probably move it to "Install Python" section.

File: README.md [hunk 2]
```diff
@@ -35,6 +35,37 @@ Eland also provides tools to upload trained machine learning models from common
 
 ## Getting Started
 
+### Install Python
```
[the remark this answers is not in the document]
Comment (j-bennet) [reply README.md:38]: It's true, the internet is full of such guides. But the user still has to find a good one, one that's short and clear enough, and this creates an adoption barrier.

Sure, Eland is a data science tool, and data scientists are familiar with Python, Pandas, Numpy, Scipy, etc. However, Eland is also a recommended tool to import an ML model into Elasticsearch, and all Elasticsearch users don't necessarily come from a Python background.

With this section, I attempted to make the guide more comprehensive, and more accessible to people that are new to the Python ecosystem. I'd rather keep it, unless you have a strong objection to it.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (j-bennet) [reply README.md]: Moved.

File: README.md [hunk 3]
```diff
@@ -35,6 +35,43 @@ Eland also provides tools to upload trained machine learning models from common
...
+```shell
+xcode-select --install
+```
+
+Then, Python 3 can be installed with [homebrew](https://brew.sh/):
+
+```shell
+brew install python
+```
+
```
Comment (davidkyle) [inline README.md:63]:
```suggestion
```

This is covered above where readers are instructed to download python from https://www.python.org/downloads/

File: README.md [hunk 4]
```diff
@@ -35,6 +35,43 @@ Eland also provides tools to upload trained machine learning models from common
...
+preinstalled. To verify the Python version, run your preferred terminal application and type in the command:
+
+```shell
+python --version
+```
+
+* If `python` command is not available, see if you have `python3`.
+* If you have both `python` and `python3`, pick the one with higher version.
+* If you have either `python` or `python3, but the version is below 3.7, [download and install newer Python](https://www.python.org/downloads/).
+* If you don't have `python` or `python3`, [download and install Python](https://www.python.org/downloads/).
```
Comment (davidkyle) [inline README.md:50]:
```suggestion
* If you are using Eland to import NLP model version 3.10 or lower of Python is required. 
* If you don't have `python` or `python3`, [download and install Python](https://www.python.org/downloads/).
```

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidkyle) [reply README.md:38]: You make a good case 👍

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by davidkyle** [comment]
I opened #505 to investigate the linting failures

**Comment by j-bennet** [comment]
I applied the suggestions @davidkyle . Will you merge, since I have no write access?

File: README.md [hunk 5]
```diff
@@ -47,14 +78,20 @@ Eland can also be installed from [Conda Forge](https://anaconda.org/conda-forge/
 $ conda install -c conda-forge eland
 ```
 
+By default, Eland doesn't install PyTorch. If you plan to import NLP models, install `pytorch` extras:
```
[the remark this answers is not in the document]
Comment (davidkyle) [reply README.md:82]: Can you add this same text to the installation guide please.

https://github.com/elastic/eland/blob/main/docs/guide/installation.asciidoc

**Pull request closed without merging**